File: go/anagram.go
package main

func isAnagram(s, t string) bool {
	abc := 26

	charSetS := make([]int, abc)
	charSetT := make([]int, abc)

	for _, r := range s {
		charSetS[charPosition(r)]++
	}

	for _, r := range t {
		charSetT[charPosition(r)]++
	}

	if len(charSetS) != len(charSetT) {
		return false
	}

	for i, c := range charSetS {
		if charSetT[i] != c {
			return false
		}
	}

	return true
}

func charPosition(r rune) int {
	if r >= 'a' && r <= 'z' {
		return int(r - 'a')
	} else if r >= 'A' && r <= 'Z' {
		return int(r - 'A')
	}

	return -1
}

File: zig/src/anagram.zig
const std = @import("std");

pub fn isAnagram(s: []const u8, t: []const u8) !bool {
    if (t.len != s.len) {
        return false;
    }

    var char_set_s: [26]i32 = undefined;
    var char_set_t: [26]i32 = undefined;

    for (s) |c| {
        const idx = c - 'a';
        char_set_s[idx] += 1;
    }

    for (t) |c| {
        const idx = c - 'a';
        char_set_t[idx] += 1;
    }

    for (char_set_s, 0..) |count, i| {
        if (count != char_set_t[i]) {
            return false;
        }
    }

    return true;
}
